packages/medusa-plugin-sendgrid/src/services/sendgrid.js

```diff
@@ -216,6 +218,9 @@ class SendGridService extends NotificationService {
       case "order.refund_created":
         return this.options_.order_refund_created_template
       default:
+        if (templates.find((template) => event == template)) {
+          return this.options_[event]
+        }
```

Comment:
todo: I believe that you can replace the entire switch with this loop instead

```ts
const templates = Object.keys(this.options_ ?? {})
return templates.find((template) => event.toLowerCase() === template.toLowerCase())
```

wdyt?

Reply:
maybe..  we'll need to rewrite it like so to maintain backward compatibility
```
 const templates = Object.keys(this.options_ ?? {})
    const key = templates.find(
      (template) =>
        event.toLowerCase().toLowerCase().replace(".", "_") === template ||
        `${event.toLowerCase().toLowerCase().replace(".", "_")}_template` ==
          template
    )
    return key ? this.options_[key] : key
 ```
 wdyt

Reply:
btw thanks for taking the time to review this .,. much appreciated

Reply:
not sure if this is still and issue. The . to _ replacement is necessary because the templates in the options have been defined like that. where as the event names have. This was done to ensure that existing logic continues to work in addition to the new logic